tests/transactional/verify_undelete_snapshots.pm

```diff
@@ -0,0 +1,67 @@
+# Maintainer: QA SLE Functional YaST <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use transactional qw(rpmver get_utt_packages trup_call);
+
+
+# Verify that essential snapshots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+sub run {
+    select_console 'root-console';
+
+    my @snapshots_before = (split /\n/, script_output('snapper list --disable-used-space'));
+    my $current_snapshot_before;
+    foreach (@snapshots_before) {
+        if ($_ =~ /(?<current_snapshot_before>^\d+)\*/) { $current_snapshot_before = $+{current_snapshot_before} }
+    }
+
+    # Create some snapshots with a transactional update snapshot in the middle.
+    record_info "Check undel", "Snapshot #1 - Check that essential snapshots cannot be deleted, 
+                                creates snapshots and installs a package";
+    assert_script_run("snapper create -d \"Disposable snapshot #1\"");
+    get_utt_packages;
+    trup_call "ptf install" . rpmver('security');
+    assert_script_run("snapper create -d \"Disposable snapshot #2\"");
+
+    my @snapshots_after_update = (split /\n/, script_output('snapper list --disable-used-space'));
+    my $next_snap_after_update;
+    my $current_snap_after_update;
+    foreach (@snapshots_after_update) {
+        if ($_ =~ /(?<current_snap>^\d+)\-/) { $current_snap_after_update = $+{current_snap} }
+        if ($_ =~ /(?<next_snap>^\d+)\+/)    { $next_snap_after_update    = $+{next_snap} }
+    }
+    die('Current snapshot was not marked with -') if (!length $current_snap_after_update);
+    die('New snapshot not marked with +')         if (!length $next_snap_after_update);
+    die unless ($current_snapshot_before == $current_snap_after_update);
+
+    my $last_snapshot        = $snapshots_after_update[-1];
+    my $last_snapshot_number = (split ' ', $last_snapshot)[0];
+    my @expected             = ('current system', 'currently mounted', 'next to be mounted');
+    #  The line below simulates a distracted admin trying to delete all snapshots...
+    my @delete_response = (split /\n/, script_output("snapper delete 0-$last_snapshot_number 2>&1"));
+    my @snap_numbers    = ("0", $current_snap_after_update, $next_snap_after_update);
+    for (my $line = 0; $line <= $#expected; $line++) {
+        record_info("$expected[$line]", "Snapper delete response: $delete_response[$line]");
+        if ($delete_response[$line] !~ /^Cannot delete snapshot $snap_numbers[$line] since it is the $expected[$line].*/) {
+            die("Deletion of essential snapshot not prevented ?
+                 Expected: \"Cannot delete...$expected[$line]...\" got: \"$delete_response[$line]\"");
+        }
+    }
+    my @snapshots_after = (split /\n/, script_output('snapper list --disable-used-space'));
+    my $output_lines    = 5;                                                                  # number of lines for 3 snapshots +header
```

Comment:
this comment for some reason is visualized too on the right, maybe is github that displays like that, if not, perhaps is better on top of the assignation. Ignore this if doesn't make sense, people use different editors and sometimes requires some configuration but probably tidy messed up this :)

Reply:
This is due to the tidy script :(